src/gurt/debug.c

```diff
@@ -595,8 +595,8 @@ d_log_init(void)
 	if (log_file != NULL && log_file_pid_append != NULL) {
 		if (strcmp(log_file_pid_append, "0") != 0) {
 			/* Append pid to log file. */
-			asprintf(&buffer, "%s%d", log_file, getpid());
-			if (buffer == NULL)
+			rc = asprintf(&buffer, "%s%d", log_file, getpid());
+			if (buffer == NULL || rc == -1)
```

Comment:
Not sure we care but the man page makes it sound as if we can get -1 for errors not related to memory allocation.  In such case, -DER_NOMEM isn't strictly accurate.